tests/functional/test_data_completeness.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from botocore.session import Session
+from botocore.loaders import Loader
+from botocore.exceptions import DataNotFoundError
+
+
+class TestDataTypes(object):
```

Comment:
I am open and flexible.  Just want to make sure we are on the same page about the pros and cons.

If I understand correctly, we agree that the class-based approach is (slightly) better; should the need arises, we can adapt to it in spite of inconsistency.

Deduction 1: In the long term, inconsistency will eventually be there anyway, how about start to adapt to the future-proof approach now?

Deduction 2: At a hindsight, if we rewind back to day 1, perhaps we would write the test cases in class-based flavor, so there wouldn't be such inconsistency in the first place. (I suspect that the flatten-function flavor is influenced by the [nose 101 document](http://nose.readthedocs.org/en/latest/writing_tests.html), but 101 document typically focuses only on features, it does not necessarily guide us how to organize the code.)

Last but not the least, if we stick with current flatten-function flavor (which is fine), chances are, we won't bother to refactor old runnable test cases when we need to add more test cases in future because "If it ain't broken, why fix it?", and then even the new test cases are less likely to choose the class-based flavor to violate the convention in the same file.

By the way, in this specific case, both functions use a loader instance independently. We could choose to make that loader instance as a member in this class to be shared by the 2 methods. That should make the test cases run a bit faster. Will that be better reason alone, on top of all above discussion?